shared/src/main/scala/scala/xml/factory/XMLLoader.scala

```diff
@@ -45,7 +45,11 @@ trait XMLLoader[T <: Node] {
   }
 
   /* Override this to use a different SAXParser. */
-  def parser: SAXParser = parserInstance.get
+  def parser: SAXParser = {
+    val p = parserInstance.get
+    p.reset()
```

Comment:
Auto-reset on get leads to surprising changes in behavior, e.g. non-persistence of the settings on `XMLLoader.reader`, as witnessed by the changes to the test above.
Better place for auto-reset seems to be `FactoryAdapter.loadDocument()`, but it will then apply to all readers, not just the default one...
Better still would be - in my opinion - not to reuse parsers, readers and factory adapters and not to use thread-locals.

Reply:
So revert https://github.com/scala/scala-xml/pull/176?

Reply:
I am not going to campaign for reverting it,  but I am not a fan ;)